portfolio/src/main/webapp/funfacts.html

```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <title>Hadley Shapland</title>
+    <link rel="stylesheet" href="style.css">
+    <script src="script.js"></script>
+  </head>
+
+  <body>
+        <!-- SIDEBAR -->
+        <div class="sidenav">
+            <a href="index.html">Home</a>
+            <a href="background.html">Background</a>
+            <a href="experience.html">Experience</a>
+            <a href="funfacts.html">Fun Facts</a>
+        </div>
+        <!-- END SIDEBAR -->
```

Comment:
The sidebar seems to be duplicated in each file.

In general, we want to avoid duplication since it increases maintenance burdens and is highly error prone (someone editing the code later on will have to find and edit all the copies correctly, but will often mess one up without noticing; Over time this gets compounded and leads to systems that are impossible to work with).

It will be a bit more challenging, but could you find a way to share this logic between files? There's several ways to go about it, traditionally one would use an iframe, but now (html5) the simplest approach is probably a link/rel. My personal preference would be to create this in a script (and then share the script), using dom manipulation to build the nav programatically (though this isn't usually how most sites do it).

You should find how to do this with a search like "include html from another file", but also check out w3schools dom manipulation totorial if you're interested in the latter approach.

If you'd like to work on this in a seperate pull that's fine.

Reply:
Good point, I'm going to work on this in a new branch to make sure that it works correctly.